pkg/route/engine.go

```diff
@@ -901,6 +901,17 @@ func (engine *Engine) HasServer(name string) bool {
 	return engine.protocolSuite.Get(name) != nil
 }
 
+// SetBasePath To modify basePath
+func (engine *Engine) SetBasePath(basePath string) {
```

Comment:
Add `SetBasePath` method to Engine, will make the group scene gets complicated（cause it may be called anywhere & anytime）. If it is for initiation of a context-aware hertz instance, it is better put it to the option of server.(eg. https://github.com/cloudwego/hertz/blob/develop/pkg/app/server/option.go#L198)